tools/clang/test/HLSLFileCheck/d3dreflect/rdat_mintarget/sm66_derivatives_lib_6_6.ll

```diff
@@ -0,0 +1,218 @@
+; RUN: %dxilver 1.6 | %D3DReflect %s | FileCheck %s -check-prefixes=RDAT,RDAT16
+
+target datalayout = "e-m:e-p:32:32-i1:32-i8:32-i16:32-i32:32-i64:64-f16:32-f32:32-f64:64-n8:16:32:64"
+target triple = "dxil-ms-dx"
+
+%dx.types.Handle = type { i8* }
+%dx.types.ResourceProperties = type { i32, i32 }
+%struct.RWByteAddressBuffer = type { i32 }
+
+@"\01?BAB@@3URWByteAddressBuffer@@A" = external constant %dx.types.Handle, align 4
+
+; Ensure min shader target incorporates optional features used
+
+; RDAT: FunctionTable[{{.*}}] = {
+
+; SM 6.6+
+
+;/////////////////////////////////////////////////////////////////////////////
+; ShaderFeatureInfo_DerivativesInMeshAndAmpShaders (0x1000000) = 16777216
+; OptFeatureInfo_UsesDerivatives (0x0000010000000000) = FeatureInfo2: 256
+
+; OptFeatureInfo_UsesDerivatives Flag used to indicate derivative use in
+; functions, then fixed up for entry functions.
+; Val. ver. 1.8 required to recursively check called functions.
+
+; RDAT-LABEL: UnmangledName: "deriv_in_func"
+; RDAT: FeatureInfo1: 0
+; OptFeatureInfo_UsesDerivatives (0x0000010000000000) = FeatureInfo2: 256
+; RDAT18:   FeatureInfo2: 256
+; Old: deriv use not tracked
+; RDAT16:   FeatureInfo2: 0
+; RDAT17:   FeatureInfo2: 0
+; Pixel(0), Compute(5), Library(6), Mesh(13), Amplification(14), Node(15) = 0xE061 = 57441
+; RDAT18: ShaderStageFlag: 57441
+; Old would not report Compute, Mesh, Amplification, or Node compatibility.
+; Pixel(0), Library(6) = 0x41 = 65
+; RDAT16: ShaderStageFlag: 65
+; RDAT17: ShaderStageFlag: 65
+; MinShaderTarget: (Library(6) << 16) + (SM 6.0 ((6 << 4) + 0)) = 0x60060 = 393312
+; RDAT17: MinShaderTarget: 393312
+; RDAT18: MinShaderTarget: 393312
+; Old: Didn't set min target properly for lib function
+; RDAT16: MinShaderTarget: 393318
```

Comment:
The issue is that I had to split these into three files for different versions, and they needed the identical checks that the HLSL file had.  I wanted to preserve the checks as-is so copy-paste between would work.  If I tailor each to each version, it would be a lot harder to edit and maintain.  I wish there was a way to do multiple RUN lines with preprocessing and ifdef/macro expansion on the source for .ll tests.  Then I could use an include file for common checks and/or code.